### PR TITLE
Feat/docker registry

### PR DESCRIPTION
**Pull request by nhaghighat**
## Description

This PR add docker registry deployment to orchest with SSL.
The Kaniko should be created with additional argument :
`--registry-certificate=/etc/ssl/docker/ca.crt`
and with added mounted secret.
in volumes section:
`
        - name: registry-tls
          secret:
            secretName: registry-tls-secret
`

in volumeMounts section:
`            - mountPath: /etc/ssl/docker
              name: registry-tls-secret
              readOnly: true
`

## Checklist

- [ ] The PR branch is set up to merge into `k8s` instead of `master`.

### CODE REVIEW

**Review by fruttasecca — APPROVED** [review]
Can't really comment much on the yaml files, they look correct. I have been developing against this PR/deployment and I haven't met issues.

About the third-party repositories which we store in our repo, is there a reason to not have those as submodules pointing to a specific commit? Aside from the extra step of pulling submodules

**Comment by fruttasecca** [comment]
Forgot: can we make the registry service map 443:5000? That's the port that most clients work with by default

**Comment by nhaghighat** [comment]
> Can't really comment much on the yaml files, they look correct. I have been developing against this PR/deployment and I haven't met issues.
> 
> About the third-party repositories which we store in our repo, is there a reason to not have those as sub modules pointing to a specific commit? Aside from the extra step of pulling sub modules

The main reason that I did not use sub modules, was that sometimes the third-parties are living as part of other git repositories, and it is not possible to have a submodule that points to subdirectories of a git repo.

File: deploy/Makefile [hunk 1]
```diff
@@ -7,11 +7,74 @@ DEBUG := ${if ${DEBUG},--debug --dry-run,}
...
+
+# Orchest deployment configs
+ORCHEST_ENABLE_CONFIG := ${ORCHEST_API_ENABLE_CONFIG} \
+						 ${AUTH_SERVER_ENABLE_CONFIG} \
+						 ${AUTH_SERVER_ENABLE_CONFIG} \
+						 ${FILE_MANAGER_ENABLE_CONFIG} \
+						 ${ORCHEST_WEBSERVER_ENABLE_CONFIG} \
+						 ${POSTGRES_ENABLE_CONFIG} \
+						 ${RABBITMQ_ENABLE_CONFIG} \
+						 ${UPDATE_SERVER_ENABLE_CONFIG}
```
Comment (yannickperrenet) [inline deploy/Makefile:74]: I guess this means that the `update-server` is started when Orchest is deployed? (Fine for now, just checking.)

Comment (nhaghighat) [reply deploy/Makefile:74]: Indeed, and we have the option to disable it later

File: deploy/thirdparty/cert-manager/cert-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,5266 @@
+# Copyright 2022 The cert-manager Authors.
```
[the remark this answers is not in the document]
Comment (nhaghighat) [reply deploy/thirdparty/cert-manager/cert-manager.yaml]: Indeed, the additional configurations are in cluster-resources in helm directory.

File: deploy/helm/templates/_cluster-registry.tpl [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+*/}}
+{{- define "library.cluster.registry.certificate" -}}
+{{- .Values.registryCertificate | default "registry-certificate" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Get the dns of docker-registry.
+*/}}
+{{- define "library.cluster.registry.dns" -}}
+{{ printf "%s.%s.svc.cluster.local" .Values.registry.name .Release.Namespace }}
```
Comment (yannickperrenet) [inline deploy/helm/templates/_cluster-registry.tpl:12]: We don't set this `.Values.registry.name` do we? Couldn't find it.

Comment (nhaghighat) [reply deploy/helm/templates/_cluster-registry.tpl:12]: It is seted here:
https://github.com/orchest/orchest/pull/677/files#diff-ca41056a0340e0139d50c70288e712c894b470842211c94554a1a493ed0164f7R35

Comment (yannickperrenet) [reply deploy/helm/templates/_cluster-registry.tpl:12]: Thanks. Missed that one.

**Review by yannickperrenet — APPROVED** [review]
Awesome stuff! :muscle:

**Pull request merged**